home/home_2.2/main.swift

```diff
@@ -6,87 +6,103 @@ struct Pixel: Equatable {
     var b: Int
 }
 
-let WIDTH = 5
-let HEIGHT = 5
-
-func readImage(filename: String, outputFilename: String) -> [[Pixel]]? {
-    guard let input = try? String(contentsOfFile: filename) else {
-        print("Unable to open input file.")
-        return nil
-    }
-    
-    var image: [[Pixel]] = []
-    let lines = input.components(separatedBy: .newlines)
+class Image {
+    static let width = 5
+    static let height = 5
+    var pixels: [[Pixel]]
     
-    guard lines.count == HEIGHT else {
-        print("Input file should contain exactly \(HEIGHT) lines.")
-        try? "Input file should contain exactly \(HEIGHT) lines.".write(toFile: outputFilename, atomically: true, encoding: .utf8)
-        return nil
+    init(pixels: [[Pixel]]) {
+        self.pixels = pixels
     }
-
-    for (row, line) in lines.enumerated() {
-        let parts = line.components(separatedBy: " ")
-        guard parts.count == WIDTH else {
-            print("Input file should contain exactly \(WIDTH) pixels per line.")
+    
+    static func readInfo(filename: String, outputFilename: String) -> Image? {
```

Comment:
minor: this function seems to be big enough. consider some ways to split and refactor it